src/Rings/mpoly-affine-algebras.jl

```diff
@@ -931,25 +941,29 @@ julia> is_normal(A)
 true
 ```
 """
-@attr Bool function is_normal(A::MPolyQuoRing)
-  @req coefficient_ring(A) isa AbstractAlgebra.Field "The coefficient ring must be a field"
-  @req !(base_ring(A) isa MPolyDecRing) "Not implemented for quotients of decorated rings"
-
-  I = A.I
-  # TODO remove old1 & old2 once new Singular jll is out
-# QUESTION: How old is this? Has this been fixed yet?
-  old1 = Singular.libSingular.set_option("OPT_REDSB", false)
-  old2 = Singular.libSingular.set_option("OPT_RETURN_SB", false)
-  f = Singular.LibNormal.isNormal(singular_generators(I))::Int
-  Singular.libSingular.set_option("OPT_REDSB", old1)
-  Singular.libSingular.set_option("OPT_RETURN_SB", old2)
-  return Bool(f)
+function is_normal(A::MPolyQuoRing; check::Bool=true)
+  return get_attribute!(A, :is_normal) do
+    @req coefficient_ring(A) isa AbstractAlgebra.Field "Only implemented if coefficient ring is a field"
+    @req is_perfect(coefficient_ring(A)) "Only implemented if coefficient ring is a perfect field"
+    @req !(base_ring(A) isa MPolyDecRing) "Not implemented for quotients of decorated rings"
+    !check || is_reduced(A) || return false
+
+    I = A.I
+    # TODO remove old1 & old2 once new Singular jll is out
+    # QUESTION: How old is this? Has this been fixed yet?
```

Comment:
@hannes14 : Do you have an idea whether this has already propagated up or not?
                         This would be the moment to get rid of the workaround, if it is not needed any more.